src/Chapter2.hs

```diff
@@ -858,7 +883,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind (x:xs) = (rewind xs)++[x]
```

Comment:
In a procedural language I would swap the elements from the outer ends on inwards with a for loop.  I wonder whether this implementation will scatter the list all over the stack?